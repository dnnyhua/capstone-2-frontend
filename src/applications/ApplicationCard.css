.ApplicationCard {
    width: 1000px;
    height: max-content;
    padding: 10px;
    margin: 10px;
    background-color: rgb(68, 68, 68);
}

.hired {
    background-color: rgb(94, 212, 173);
}

.ApplicationCard-img {
    opacity: 1.0;
    border-radius: 50%;
}

.ApplicationCard-body {
    display: flex;
    position: relative;
    flex-direction: row;
    align-items: center;
    width: 100%;

}

.ApplicationCard-hireBtn {
    display: flex;
    flex-direction: row;
    position: absolute;
    top: 77%;
    left: 85%;
    font-weight: bold;


}


.ApplicationCard-rejectBtn {
    display: flex;
    flex-direction: row;
    position: absolute;
    top: 77%;
    left: 92%;
    font-weight: bold;
}


.acknowledgmentModal {
    opacity: 0;
    visibility: hidden;
    display: none;
    position: fixed;
    align-items: center;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);
    transition: opacity 0.3s ease, visibility 0.3s ease;
    align-content: center;
    z-index: 100;
}

.acknowledgmentModal.show {
    opacity: 1;
    visibility: visible;
    display: flex;
    align-items: center;

}


.modal-title {
    color: black
}

.modal-content {
    background: azure;
    padding: 10px;
}

p {
    color: black
}