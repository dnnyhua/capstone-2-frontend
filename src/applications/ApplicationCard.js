import React, { useEffect, useState, useContext } from "react";
import './ApplicationCard.css'
import Api from "../api";
import { useNavigate } from "react-router-dom";
import GlobalContext from "../helper/GlobalContext";

const ApplicationCard = ({ jobId, walkerId, status, firstName, lastName, ratePer30min, getApplications }) => {
    const { getCurrUserData } = useContext(GlobalContext);

    const [isRejected, setIsRejected] = useState(false);
    const [currStatus, setCurrStatus] = useState(status)


    // THIS IS NOT RE_RENDERING
    async function hireWalker() {
        try {
            setCurrStatus("Hired")
            await Api.hireWalker(jobId, walkerId)
        } catch (err) {
            console.error("Error hiring walker:", err)
        }
    }

    const handleHireBtn = async () => {
        await hireWalker();
        getCurrUserData();
    }

    async function rejectWalker() {
        try {
            setIsRejected(true)
            setCurrStatus("")
            await Api.rejectWalker(jobId, walkerId)
        } catch (err) {
            console.error("Error rejecting walker: err")
        }

    }

    const handleRejectBtn = async () => {
        rejectWalker();
        getApplications();
    }

    useEffect(() => {
    }, [isRejected, setIsRejected, currStatus, getCurrUserData])


    console.log(currStatus)

    return (
        <div className={`ApplicationCard ${currStatus === "Hired" ? "hired" : ""}`} >
            <div className="ApplicationCard-body">
                <section>
                    <img className="ApplicationCard-img" src="https://static.thenounproject.com/png/5034901-200.png" />
                </section>

                <section>
                    <h5>Job Id: {jobId} </h5>
                    <h5>Walker Id: {walkerId}</h5>
                    <h5>{firstName} {lastName}</h5>
                    <h5>Rate: ${ratePer30min} / 30 min</h5>
                    <h5>Add bio here; need to update schema and seed file</h5>
                </section>

                {currStatus === 'Pending Review' ? (
                    <section>
                        <button className="btn btn-primary ApplicationCard-hireBtn" onClick={handleHireBtn} disabled={isRejected}>Hire</button>
                        <button className="btn btn-danger ApplicationCard-rejectBtn" onClick={handleRejectBtn} disabled={isRejected}>Reject</button>
                    </section>
                ) : (
                    <section>
                        <button disabled className="btn btn-secondary ApplicationCard-hireBtn" >Hire</button>
                        <button disabled className="btn btn-secondary ApplicationCard-rejectBtn">Reject</button>
                    </section>
                )}
            </div>
        </div >
    )
}

export default ApplicationCard;