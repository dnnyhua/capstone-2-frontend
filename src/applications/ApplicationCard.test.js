import React from "react";
import { render } from "@testing-library/react";
import ApplicationCard from "./ApplicationCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with profileImage", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <ApplicationCard
                firstName="Johnny"
                lastName="Bravo"
                rate='20'
                bio="i love all dogs!"
                profileImage="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSRNF8-gfvx__gaeVMxKBA2uxYZWju8XBV3_P0Rmsl_9lAcFVJCts7MZbNbE0bnARq1FFE&usqp=CAU"
            />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without profileImage", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <ApplicationCard
                firstName="Johnny"
                lastName="Bravo"
                rate='20'
                bio="i love all dogs!"
            />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});



