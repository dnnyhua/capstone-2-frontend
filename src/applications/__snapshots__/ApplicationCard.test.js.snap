// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with profileImage 1`] = `
<DocumentFragment>
  <div
    class="ApplicationCard "
  >
    <div
      class="ApplicationCard-body"
    >
      <section>
        <img
          class="ApplicationCard-img"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSRNF8-gfvx__gaeVMxKBA2uxYZWju8XBV3_P0Rmsl_9lAcFVJCts7MZbNbE0bnARq1FFE&usqp=CAU"
        />
      </section>
      <section>
        <h5>
          Job Id:  
        </h5>
        <h5>
          Walker Id: 
        </h5>
        <h5>
          Johnny Bravo
        </h5>
        <h5>
          Rate: $20 / 30 min
        </h5>
        <h5>
          About Me: i love all dogs!
        </h5>
      </section>
      <section>
        <button
          class="btn btn-secondary ApplicationCard-hireBtn"
          disabled=""
        >
          Hire
        </button>
        <button
          class="btn btn-secondary ApplicationCard-rejectBtn"
          disabled=""
        >
          Reject
        </button>
      </section>
    </div>
    <div
      class="acknowledgmentModal "
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content "
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
            >
              Acknowledgement
            </h5>
          </div>
          <div
            class="modal-body"
          >
            <p>
              Once hired, you cannot select a different walker. Are you sure you want to hire this walker?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Yes
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              No
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without profileImage 1`] = `
<DocumentFragment>
  <div
    class="ApplicationCard "
  >
    <div
      class="ApplicationCard-body"
    >
      <section>
        <img
          class="ApplicationCard-img"
        />
      </section>
      <section>
        <h5>
          Job Id:  
        </h5>
        <h5>
          Walker Id: 
        </h5>
        <h5>
          Johnny Bravo
        </h5>
        <h5>
          Rate: $20 / 30 min
        </h5>
        <h5>
          About Me: i love all dogs!
        </h5>
      </section>
      <section>
        <button
          class="btn btn-secondary ApplicationCard-hireBtn"
          disabled=""
        >
          Hire
        </button>
        <button
          class="btn btn-secondary ApplicationCard-rejectBtn"
          disabled=""
        >
          Reject
        </button>
      </section>
    </div>
    <div
      class="acknowledgmentModal "
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content "
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
            >
              Acknowledgement
            </h5>
          </div>
          <div
            class="modal-body"
          >
            <p>
              Once hired, you cannot select a different walker. Are you sure you want to hire this walker?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Yes
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              No
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
